mono/metadata/marshal.c

```diff
@@ -9128,8 +9128,8 @@ mono_marshal_get_struct_to_ptr (MonoClass *klass)
 
 	mono_marshal_load_type_info (klass);
 
-	if (klass->marshal_info->str_to_ptr)
-		return klass->marshal_info->str_to_ptr;
+	if (mono_class_get_marshal_info (klass)->str_to_ptr)
```

Comment:
If marshal_info becomes a property, it should only be retrieved once to avoid searching multiple times.